x-pack/plugins/cases/server/client/factory.ts

```diff
@@ -160,4 +159,62 @@ export class CasesClientFactory {
       attachmentService,
     };
   }
+
+  /**
+   * This function attempts to retrieve the current user's info. The first method is using the user profile api
+   * provided by the security plugin. If that fails or the session isn't found then we will attempt using authc
+   * which will not retrieve the profile uid but at least gets us the username and sometimes full name, and email.
+   *
+   * This function also forces the fields to be strings or null (except the profile uid since it's optional anyway)
+   * because the get case API expects a created_by field to be set. If we leave the fields as undefined
+   * then the resulting object in ES will just be empty and it'll fail to encode the user when returning it to the API
+   * request. If we force them to be null it will succeed.
+   */
+  private async getUserInfo(request: KibanaRequest): Promise<{
+    username: string | null;
+    full_name: string | null;
+    email: string | null;
+    profile_uid?: string;
+  }> {
+    if (!this.isInitialized || !this.options) {
+      throw new Error('CasesClientFactory must be initialized before calling create');
+    }
+
+    try {
+      const userProfile = await this.options.securityPluginStart?.userProfiles.getCurrent({
+        request,
+      });
+
+      if (userProfile != null) {
+        return {
+          username: userProfile.user.username,
```

Comment:
being a bit paranoid here but maybe we should do `userProfile.user.username ?? null`. I am not sure what is the user name for anonymous users. @azasypkin do you know if the username will be always defined?

Reply:
I believe you can count on `username` to be always present if the profile is available. We don't have user profiles for anonymous users yet, but if it changes in the future, there will be `username` as well (of the underlying anonymous "service account").

Reply:
Thanks!